### DISCUSSION TITLE
How do I remove the previous button from step 1?

### DISCUSSION THREAD

**Discussion by edu22snt** [hakimio/ngx-ui-tour#150] [Q&A] [answered]
How do I remove the previous button from step 1?

I need the previous button from step 1 not to appear, is this possible?

![step1](https://github.com/hakimio/ngx-ui-tour/assets/8997162/d6da5d47-879b-4608-967e-3ca0e89e0493)

**Comment by hakimio** [comment, marked as answer]
Normally the button has disabled styling.
Anyway, you can customize this by providing your own step template: https://hakimio.github.io/ngx-ui-tour/md-menu/Misc#custom-template

**Reply by edu22snt** [reply]
Do you have any examples made? I can't understand how to do it.
thank you very much.

**Reply by hakimio** [reply]
I have linked to the example code.
```html
<tour-step-template>
    <ng-template let-step="step">
        <my-custom-tour-step [step]="step">
        </my-custom-tour-step>
    </ng-template>
</tour-step-template>
```

**Comment by edu22snt** [comment]
Thanks very much.

Em ter., 10 de out. de 2023 às 04:29, Tomas Rimkus ***@***.***>
escreveu:

> I have linked to the example code.
>
> <tour-step-template>
>     <ng-template let-step="step">
>         <my-custom-tour-step [step]="step">
>         </my-custom-tour-step>
>     </ng-template></tour-step-template>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hakimio/ngx-ui-tour/discussions/150#discussioncomment-7237748>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACEUSKR6YELORNJNF4TL46LX6T2OBAVCNFSM6AAAAAA5V3HOZCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TEMZXG42DQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Atenciosamente;

*Eduardo Monteiro de Andrade*
Desenvolvedor Java
Tel: +55 (61) 9 8458-8881


.....Respondeu JESUS: "Eu sou o caminho a verdade e a vida. Ninguém vem ao
Pai, a não ser por mim" João 14:6......

**Comment by edu22snt** [comment]
Good morning,
Sorry, but I can't get it to work.

I put it the way you told me, but I can't understand what I should do.

![1](https://github.com/hakimio/ngx-ui-tour/assets/8997162/c6cf7e98-31a3-4b2b-9588-bfdcb3ea9336)

displays this error

![2](https://github.com/hakimio/ngx-ui-tour/assets/8997162/4c221854-8c95-4ba5-a26e-fd0096a54ec1)

I did it this way, but it doesn't change anything.

![3](https://github.com/hakimio/ngx-ui-tour/assets/8997162/469df644-fd0c-4b77-aeea-7af2f12cbdc1)

I just need to make the Voltar button disappear
*ngIf="!tourService.hasPrev(step)"

Can you help me?

**Reply by hakimio** [reply]
You have to actually implement your "custom tour step" (called `<my-custom-tour-step>` in my example). It's not provided for you.
